### PR TITLE
Allow individual Airspeed backends to be compiled out

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
Everything still compiles, and we can compile each of the backends out separately or collectively.

**Comment by tridge** [comment]
test flown on a DroneCAN airspeed and MS4525 I2C today with no issues

**Comment by MallikarjunSE** [comment]
Hey @peterbarker 
define AP_AIRSPEED_BACKEND_DEFAULT_ENABLED 1 will enable all the drivers and 0 will pick defined driver?

**Comment by peterbarker** [comment]
> Hey @peterbarker define AP_AIRSPEED_BACKEND_DEFAULT_ENABLED 1 will enable all the drivers and 0 will pick defined driver?

Not quite.  You can still turn individual backends off if you set the default to 1, and you in the second case it's truth that matters, not definition, when turning on a backend if the default has been set to 0.

File: Tools/scripts/build_options.py [hunk 1]
```diff
@@ -138,6 +137,16 @@ def __init__(self,
     Feature('GPS Drivers', 'SBF', 'AP_GPS_SBF_ENABLED', 'Enable SBF GPS', 0, None),
     Feature('GPS Drivers', 'SIRF', 'AP_GPS_SIRF_ENABLED', 'Enable SiRF GPS', 0, None),
 
+    Feature('Airspeed Drivers', 'analog', 'AP_GPS_ANALOG_ENABLED', 'Enable analog Airspeed', 0, ['AP_AIRSPEED_ENABLED']),
```
Comment (tridge) [inline Tools/scripts/build_options.py:140]: why do we have "GPS" in airspeed defines?

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by peterbarker** [comment]
```
pbarker@bluebottle:~/rc/ardupilot(pr/recursion-thing)$ cat /tmp/some.out | grep -i saves
Disabling analog(AP_AIRSPEED_ANALOG_ENABLED) on copter saves 288 bytes
Disabling analog(AP_AIRSPEED_ANALOG_ENABLED) on plane saves 288 bytes
Disabling analog(AP_AIRSPEED_ANALOG_ENABLED) on rover saves 280 bytes
Disabling analog(AP_AIRSPEED_ANALOG_ENABLED) on antennatracker saves -24 bytes
Disabling analog(AP_AIRSPEED_ANALOG_ENABLED) on sub saves 280 bytes
Disabling analog(AP_AIRSPEED_ANALOG_ENABLED) on blimp saves -16 bytes
Disabling ASP5033(AP_AIRSPEED_ASP5033_ENABLED) on copter saves 992 bytes
Disabling ASP5033(AP_AIRSPEED_ASP5033_ENABLED) on plane saves 992 bytes
Disabling ASP5033(AP_AIRSPEED_ASP5033_ENABLED) on rover saves 1008 bytes
Disabling ASP5033(AP_AIRSPEED_ASP5033_ENABLED) on antennatracker saves -24 bytes
Disabling ASP5033(AP_AIRSPEED_ASP5033_ENABLED) on sub saves 1008 bytes
Disabling ASP5033(AP_AIRSPEED_ASP5033_ENABLED) on blimp saves -24 bytes
Disabling DLVR(AP_AIRSPEED_DLVR_ENABLED) on copter saves 928 bytes
Disabling DLVR(AP_AIRSPEED_DLVR_ENABLED) on plane saves 928 bytes
Disabling DLVR(AP_AIRSPEED_DLVR_ENABLED) on rover saves 936 bytes
Disabling DLVR(AP_AIRSPEED_DLVR_ENABLED) on antennatracker saves -24 bytes
Disabling DLVR(AP_AIRSPEED_DLVR_ENABLED) on sub saves 936 bytes
Disabling DLVR(AP_AIRSPEED_DLVR_ENABLED) on blimp saves -16 bytes
Disabling MS4525(AP_AIRSPEED_MS4525_ENABLED) on copter saves 1704 bytes
Disabling MS4525(AP_AIRSPEED_MS4525_ENABLED) on plane saves 1704 bytes
Disabling MS4525(AP_AIRSPEED_MS4525_ENABLED) on rover saves 1712 bytes
Disabling MS4525(AP_AIRSPEED_MS4525_ENABLED) on antennatracker saves -24 bytes
Disabling MS4525(AP_AIRSPEED_MS4525_ENABLED) on sub saves 1712 bytes
Disabling MS4525(AP_AIRSPEED_MS4525_ENABLED) on blimp saves -16 bytes
Disabling MS5525(AP_AIRSPEED_MS5525_ENABLED) on copter saves 1672 bytes
Disabling MS5525(AP_AIRSPEED_MS5525_ENABLED) on plane saves 1672 bytes
Disabling MS5525(AP_AIRSPEED_MS5525_ENABLED) on rover saves 1680 bytes
Disabling MS5525(AP_AIRSPEED_MS5525_ENABLED) on antennatracker saves -24 bytes
Disabling MS5525(AP_AIRSPEED_MS5525_ENABLED) on sub saves 1680 bytes
Disabling MS5525(AP_AIRSPEED_MS5525_ENABLED) on blimp saves -16 bytes
Disabling MSP(AP_AIRSPEED_MSP_ENABLED) on copter saves 560 bytes
Disabling MSP(AP_AIRSPEED_MSP_ENABLED) on plane saves 560 bytes
Disabling MSP(AP_AIRSPEED_MSP_ENABLED) on rover saves 552 bytes
Disabling MSP(AP_AIRSPEED_MSP_ENABLED) on antennatracker saves -8 bytes
Disabling MSP(AP_AIRSPEED_MSP_ENABLED) on sub saves 552 bytes
Disabling MSP(AP_AIRSPEED_MSP_ENABLED) on blimp saves 0 bytes
Disabling NMEA(AP_AIRSPEED_NMEA_ENABLED) on copter saves -72 bytes
Disabling NMEA(AP_AIRSPEED_NMEA_ENABLED) on plane saves -72 bytes
Disabling NMEA(AP_AIRSPEED_NMEA_ENABLED) on rover saves 880 bytes
Disabling NMEA(AP_AIRSPEED_NMEA_ENABLED) on antennatracker saves -24 bytes
Disabling NMEA(AP_AIRSPEED_NMEA_ENABLED) on sub saves 880 bytes
Disabling NMEA(AP_AIRSPEED_NMEA_ENABLED) on blimp saves -16 bytes
Disabling SDP3X(AP_AIRSPEED_SDP3X_ENABLED) on copter saves 2312 bytes
Disabling SDP3X(AP_AIRSPEED_SDP3X_ENABLED) on plane saves 2312 bytes
Disabling SDP3X(AP_AIRSPEED_SDP3X_ENABLED) on rover saves 2312 bytes
Disabling SDP3X(AP_AIRSPEED_SDP3X_ENABLED) on antennatracker saves -24 bytes
Disabling SDP3X(AP_AIRSPEED_SDP3X_ENABLED) on sub saves 2312 bytes
Disabling SDP3X(AP_AIRSPEED_SDP3X_ENABLED) on blimp saves -16 bytes
Disabling UAVCAN(AP_AIRSPEED_UAVCAN_ENABLED) on copter saves 2448 bytes
Disabling UAVCAN(AP_AIRSPEED_UAVCAN_ENABLED) on plane saves 2464 bytes
Disabling UAVCAN(AP_AIRSPEED_UAVCAN_ENABLED) on rover saves 2464 bytes
Disabling UAVCAN(AP_AIRSPEED_UAVCAN_ENABLED) on antennatracker saves 1856 bytes
Disabling UAVCAN(AP_AIRSPEED_UAVCAN_ENABLED) on sub saves 2464 bytes
Disabling UAVCAN(AP_AIRSPEED_UAVCAN_ENABLED) on blimp saves 352 bytes
Disabling AIRSPEED(AP_AIRSPEED_ENABLED) on copter saves 17512 bytes
Disabling AIRSPEED(AP_AIRSPEED_ENABLED) on plane saves 19624 bytes
Disabling AIRSPEED(AP_AIRSPEED_ENABLED) on rover saves 18496 bytes
Disabling AIRSPEED(AP_AIRSPEED_ENABLED) on antennatracker saves 2440 bytes
Disabling AIRSPEED(AP_AIRSPEED_ENABLED) on sub saves 16968 bytes
Disabling AIRSPEED(AP_AIRSPEED_ENABLED) on blimp saves 2408 bytes
pbarker@bluebottle:~/rc/ardupilot(pr/recursion-thing)$ 
```

File: Tools/scripts/build_options.py (hunk 1 above)
Comment (peterbarker) [reply Tools/scripts/build_options.py]: ... because copy/paste/adjust can sometimes go very, very wrong :-)

Fixed.

**Review by tridge — APPROVED** [review]

**Pull request merged**